File: Python/Pc.py
from Processor import Processor
from Disk import Disk
from Ram import Ram

class Pc():

    #private attributes
    __attProcie = Processor()
    __attDisk = Disk()
    __attRam = Ram()
    __totalPrice = 0

    #Constructor
    def __init__(self, attProcie = Processor(), attDisk = Disk(), attRam = Ram(), totalPrice = 0):
        self.__attProcie = attProcie
        self.__attDisk = attDisk
        self.__attRam = attRam
        self.__totalPrice = totalPrice

    #Setter dan Getter Processor
    def setProcessor(self, attProcie):
        self.__attProcie = attProcie

    def getProcessor(self):
        return self.__attProcie
    
    #Setter dan Getter Disk
    def setDisk(self, attDisk):
        self.__attDisk = attDisk

    def getDisk(self):
        return self.__attDisk

    #Setter dan Getter Ram
    def setRam(self, attRam):
        self.__attRam = attRam

    def getDisk(self):
        return self.__attRam

    #Setter dan Getter Total Price
    def setTotalPrice(self):
        self.__totalPrice = self.__attProcie.getPrice() + self.__attDisk.getPrice() + self.__attRam.getPrice()

    def getTotalPrice(self):
        return self.__totalPrice

    #Print data Pc
    def printPcData(self):
        print("__My PC Build Plan___________________")
        print("[Processor]")
        print("Name : " + str(self.__attProcie.getName()))
        print("Price : RP" + str(self.__attProcie.getPrice()))
        print("")
        print("[Disk]")
        print("Type : " + str(self.__attDisk.getType()))
        print("Capacity : " + str(self.__attDisk.getCapacity()))
        print("Price : RP" + str(self.__attDisk.getPrice()))
        print("")
        print("[RAM]")
        print("Capacity : " + str(self.__attRam.getCapacity()))
        print("Price : RP" + str(self.__attRam.getPrice()))
        print("_____________________________________")
        print("Total Build Price : RP" + str(self.getTotalPrice()))

File: Java/Pc.java
public class Pc {

    /* Private Attribute */
    private Processor attProcie;
    private Disk attDisk;
    private Ram attRam;
    private int totalPrice;

    /* Constuctor */
    public Pc() {
        
    }

    public Pc(Processor attProcie, Disk attDisk, Ram attRam, int totalPrice) {
        this.attProcie = attProcie;
        this.attDisk = attDisk;
        this.attRam = attRam;
        this.totalPrice = totalPrice;
    }

    /* Setter dan Getter Processor */
    public void setProcessor(Processor attProcie) {
        this.attProcie = attProcie;
    }

    public Processor getProcessor() {
        return this.attProcie;
    }

    /* Setter dan Getter Disk */
    public void setDisk(Disk attDisk) {
        this.attDisk = attDisk;
    }

    public Disk getDisk() {
        return this.attDisk;
    }

    /* Setter dan Getter Ram */
    public void setRam(Ram attRam) {
        this.attRam = attRam;
    }

    public Ram getRam() {
        return this.attRam;
    }

    /* Setter dan Getter Total Price */
    public void setTotalPrice() {
        this.totalPrice = this.attDisk.getPrice() + this.attProcie.getPrice() + this.attRam.getPrice();
    }

    public int getTotalPrice() {
        return totalPrice;
    }

    public void printPcData() {
        System.out.println("__My PC Build Plan___________________");
        System.out.println("[Processor]");
        System.out.println("Name : " + this.attProcie.getName());
        System.out.println("Price : RP" + this.attProcie.getPrice());
        System.out.println("");
        System.out.println("[Disk]");
        System.out.println("Type : " + this.attDisk.getType());
        System.out.println("Capacity : " + this.attDisk.getCapacity());
        System.out.println("Price : RP" + this.attDisk.getPrice());
        System.out.println("");
        System.out.println("[RAM]");
        System.out.println("Capacity : " + this.attRam.getCapacity());
        System.out.println("Price : RP" + this.attRam.getPrice());
        System.out.println("_____________________________________");
        System.out.println("Total Build Price : RP" + this.getTotalPrice());
    }

}